src/lang/lang_parser.mly

```diff
@@ -293,6 +296,25 @@ expr:
   | BEGIN exprs END                  { $2 }
   | FUN LPAR arglist RPAR YIELDS expr{ mk_fun ~pos:$loc $3 $6 }
   | LCUR exprss RCUR                 { mk_fun ~pos:$loc [] $2 }
+  | expr LBAR expr                   { let maybe = mk ~pos:$loc($1) (Var "maybe") in
```

Comment:
Again, I don't like much this syntax and this is not a proper `case` that you are using here :)

Reply:
I really do actually. Couple of reasons:
* The syntax is consistent with similar patterns in dynamic languages like Javascript. This will make adoption easier.
* The syntax is short
* The left side is implicitly scoped, which avoids unnecessary `{<expr>}`
* It is a case, `x | y` is the same as, in OCaml:
```ocaml
match x with
  | Some v -> v
  | None -> y
```

Reply:
I agree with that, but I don't think it is particularly intuitive (I had to look up in the grammar to understand the intended meaning). Do you think that it will be use that often? Otherwise, I'd rather leave something more explicit...

Reply:
I plan on using it a bunch yes 🙂

Reply:
Last question before I declare you won this one: can I have a reference where this is used in JS? Could not find it :(

Reply:
My bad it's double `||` not a single one in JS: https://pietschsoft.com/post/2008/10/14/javascript-gem-null-coalescing-using-the-or-operator

Looks like there's a new operator being worked on there, via '??`: https://medium.com/better-programming/how-to-use-the-null-coalescing-operator-in-javascript-15adefe7c655

Would you rather use this one? I prefer the `|` or `||` but following the same syntax could be beneficial in the long run.

Reply:
I find `??` much better: it makes it clear that it applies to an option value. The rationale behind `||` was that is overloaded the disjunction, but since it is written `or` in Liq it does not make much sense here.

Reply:
Let's do it!